addon-SV_ReportImprovements.xml

```diff
@@ -848,9 +848,10 @@ $0
     <modification template="report_view" modification_key="SV_ReportImprovements_report_view11" description="Add 'Join Conversation' button for users with permission" execution_order="10" enabled="1" action="str_replace">
       <find><![CDATA[<xen:if is="{$report.contentLink}">
 		<div class="sectionFooter actionList"><a href="{$report.contentLink}" target="_blank" class="button">{xen:phrase go_to_content}</a></div>]]></find>
-      <replace><![CDATA[$0
-<xen:elseif is="({$report.content_type} == 'conversation_message') and {$canJoinConversation}" />
-	<div class="sectionFooter actionList"><a href="{xen:link reports/conversation-join, $report}" class="button OverlayTrigger">{xen:phrase sv_join_conversation}</a></div>]]></replace>
+      <replace><![CDATA[<xen:if is="({$report.content_type} == 'conversation_message') and {$canJoinConversation}">
+	<div class="sectionFooter actionList"><a href="{xen:link reports/conversation-join, $report}" class="button OverlayTrigger">{xen:phrase sv_join_conversation}</a></div>
+<xen:elseif is="{$report.contentLink}" />
```

Comment:
This is fragile if something had modified the report content link bits.

I can't see aany reason not to just to a simple search/replace for ```{xen:raw $report.extraContentTemplate}``` and then inject after.

$report.contentLink should never be set when $canJoinConversation is set.

You can also push the content type check to code so canJoinConversation is only set for conversation message reports, which makes the template modification simpler.